apps/src/assetManagement/animationLibraryApi.js

```diff
@@ -90,6 +99,113 @@ export function createDefaultSpriteMetadata(listData) {
   });
 }
 
+export function buildAnimationMetadata(files) {
+  let animationMetadataByName = {};
+  let resolvedPromisesArray = [];
+  for (const [fileKey, fileObject] of Object.entries(files)) {
+    let json = fileObject['json'];
+    let png = fileObject['png'];
+    resolvedPromisesArray.push(
+      getAnimationLibraryFile(json.key)
+        .then(metadata => {
+          // Metadata contains name, frameCount, frameSize, looping, frameDelay
+          let combinedMetadata = metadata;
+          combinedMetadata['jsonLastModified'] = json.last_modified;
+          combinedMetadata['pngLastModified'] = png.last_modified;
+          combinedMetadata['version'] = png.version_id;
+          combinedMetadata[
+            'sourceUrl'
+          ] = `/api/v1/animation-library/level_animations/${png.version_id}/${
+            metadata.name
+          }.png`;
+          combinedMetadata['sourceSize'] = png.source_size;
+          animationMetadataByName[fileKey] = combinedMetadata;
+          return Promise.resolve();
+        })
+        .catch(err => {
+          return Promise.reject(err);
+        })
+    );
+  }
+  return Promise.all(resolvedPromisesArray).then(() => {
+    return animationMetadataByName;
+  });
+}
+
+export function buildAliasMap(animationMetadata) {
+  let aliasMap = {};
+
+  for (const [key, metadata] of Object.entries(animationMetadata)) {
+    let aliases = [metadata['name'].toLowerCase()];
+    if (metadata['aliases']) {
+      metadata['aliases'].map(alias => {
+        aliases.push(alias.toLowerCase());
+      });
+    }
+    aliases.map(alias => {
+      //Push name into target array, deduplicate, and sort
+      let updatedMap = {...aliasMap};
+      let addedSet = new Set(updatedMap[alias]);
+      addedSet.add(key);
+      aliasMap[alias] = [...addedSet].sort();
+    });
```

Comment:
i'm having trouble understanding what this block is doing. we aren't manipulating `updatedMap`, so i think we can get rid of that variable. 

i'm going to summarize what we're doing here so we can think through a clearer way to express it: map the array of alias strings to an array of objects, where the key is the alias name and the value is...?

Reply:
Yeah, the goal here is that we build a map where each alias is the key and then the value is an array of all the animations with that alias (stored in aliasMap). So line 148 gets the current set of animations with that alias. Then we add the new animation (key - line 149). Then the current set of animations (with the newly added animation) is added back to the alias map (line 150). Does that help? And same thing for the categories in the similar code block here.

Reply:
thank you, that helps a lot! now that i understand, a couple of thoughts/questions:

- do we need `updatedMap`? it's never mutated, so i'm unsure why we're creating it. if we do need it, i think we should clarify the variable name
- naming nit: `addedSet` doesn't communicate what's in the set. maybe something like `animationNameSet` or `uniqueAnimationNames` (feel free to workshop -- giving an idea so you know what i understand to be the intention of this variable)
- however, i'm not sure we want `addedSet` either. it might be clearer and more performant to deduplicate the `aliasMap` at the very end so that we're deduplicating once rather than re-creating the set in every iteration?

Reply:
Yeah, I just pushed my latest changes, which cover the first question here and possibly resolves the second one? I'll work on the third though - thanks!